spec/lita/handlers/api/market_spec.rb

```diff
@@ -0,0 +1,188 @@
+    order = {
+      id: order_id,
+      user_id: user.id,
+      type: type,
+      created_at: created_at
+    }.to_json
+    market.add_limit_order(order)
+    order
+  end
+
+  it { is_expected.to route_http(:get, 'market/limit_orders') }
+  it { is_expected.to route_http(:post, 'market/limit_orders') }
+  it { is_expected.to route_http(:post, 'market/market_orders') }
+
+  before do
+    ENV['MAX_LUNCHERS'] = '20'
+    assigner.add_to_lunchers('juan')
+    assigner.add_to_lunchers('pedro')
+    assigner.add_to_lunchers('oscar')
+    karmanager.set_karma(juan.id, 10)
+    karmanager.set_karma(pedro.id, 20)
+    karmanager.set_karma(oscar.id, 20)
+    assigner.add_to_current_lunchers(juan.mention_name)
+    assigner.add_to_current_lunchers(pedro.mention_name)
+    assigner.add_to_winning_lunchers(pedro.mention_name)
+  end
+
+  describe '#limit_orders' do
+    context 'not authorized' do
+      it 'responds with not autorized' do
+        response = JSON.parse(http.get('market/limit_orders').body)
+        expect(response['status']).to eq(401)
+        expect(response['message']).to eq('Not authorized')
+      end
+    end
+
+    context 'authorized' do
+      before do
+        allow_any_instance_of(Rack::Request).to receive(:params).and_return(user_id: pedro.id)
+        @limit_order = add_limit_order(SecureRandom.uuid, pedro, 'sell', Time.now)
+        @response = JSON.parse(http.get('market/limit_orders').body)
+      end
+
+      it 'includes the limit orders' do
+        expect(@response['limit_orders'].first.to_json).to eq(@limit_order)
+      end
+    end
+  end
+
+  describe 'POST limit order' do
+    context 'not authorized' do
+      it 'responds with not autorized' do
+        response = JSON.parse(http.post('market/limit_orders').body)
+        expect(response['status']).to eq(401)
+        expect(response['message']).to eq('Not authorized')
+      end
+    end
+
+    context 'authorized' do
+      context 'user didn\'t won lunch' do
+        before do
+          allow_any_instance_of(Rack::Request).to receive(:params).and_return(user_id: juan.id)
+          @response = JSON.parse(http.post do |req|
+            req.url 'market/limit_orders'
+            req.body = "{\"id\": \"#{SecureRandom.uuid}\",
+              \"user_id\": \"#{juan.id}\",
+              \"type\": \"sell\",
+              \"created_at\": \"#{Time.now}\"}"
+          end.body)
+        end
+
+        it { expect(@response['status']).to eq(403) }
+        it { expect(@response['message']).to eq('User can\'t place limit order') }
+
+        it 'should not add order to limits orders' do
+          expect(market.orders.size).to eq(0)
+        end
+      end
+
+      context 'user won lunch' do
+        before do
+          allow_any_instance_of(Rack::Request).to receive(:params).and_return(user_id: pedro.id)
+          @id = SecureRandom.uuid
+          @time = Time.now
+          @response = JSON.parse(http.post do |req|
```

Comment:
Acá nuevamente tienes las variables de instancia.

Reply:
¿La llamada `post` es lo que estás probando cierto?. Debería ir dentro de los bloques `it`.